src/main/java/org/verdictdb/core/querying/QueryExecutionPlanFactory.java

```diff
@@ -126,12 +117,61 @@ private static AggExecutionNode createAggExecutionNodeAndItsDependents(
       IdCreator idCreator, SelectQuery query) {
     AggExecutionNode node = new AggExecutionNode(idCreator, null);
     convertSubqueriesToDependentNodes(query, node);
-//    query.addOrderby(query.getOrderby());
-//  if (query.getLimit().isPresent()) selectQuery.addLimit(query.getLimit().get());
+    //    query.addOrderby(query.getOrderby());
+    //  if (query.getLimit().isPresent()) selectQuery.addLimit(query.getLimit().get());
+
+    int groupbyCount = 0;
+    for (GroupingAttribute attr : query.getGroupby()) {
+      UnnamedColumn groupByCol = findActualGroupByExpression(attr, query);
+      query.addSelectItem(
+          new AliasedColumn(
+              groupByCol, AsyncAggExecutionNode.getGroupByAlias() + (groupbyCount++)));
+    }
+
+    // dyoon: when 'HAVING' clause is present, its condition is added as a select item in the inner
```

Comment:
Can you clarify what is the inner query here? At this stage, I don't think there are such distinctions?

Reply:
I guess a more precise description would be that this adds those expressions into the select list of the original query, which then gets placed into the inner query of a later re-written query.

Reply:
Thanks. I think putting an example could be helpful because even I barely remember exact conversions now... Probably the same thing will happen to all of us.

Reply:
Sure. I will try to add an example as a comment for this one.